pkg/execution/cron/redis.go

```diff
@@ -0,0 +1,463 @@
+package cron
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/inngest/inngest/pkg/consts"
+	"github.com/inngest/inngest/pkg/enums"
+	"github.com/inngest/inngest/pkg/execution/queue"
+	"github.com/inngest/inngest/pkg/execution/state"
+	"github.com/inngest/inngest/pkg/execution/state/redis_state"
+	"github.com/inngest/inngest/pkg/logger"
+	"github.com/inngest/inngest/pkg/telemetry/metrics"
+	"github.com/oklog/ulid/v2"
+	"github.com/redis/rueidis"
+)
+
+var (
+	defaultScheduleForwardDur = 10 * time.Second
+)
+
+type RedisCronManagerOpt func(c *redisCronManagerOpt)
+
+type redisCronManagerOpt struct {
+	// jitter range provides a min and max duration for the jitter to apply to schedules
+	// which will move the scheduling time a little earlier than the actual cron schedule.
+	//
+	// we do this so we can make the actual run start as close as possible to the actual cron schedule.
+	jitterMin time.Duration
+	jitterMax time.Duration
+
+	// we need to move the time upwards on finding the next time due to how we use jitter
+	// to push the time back a little to coordinate the run start timing.
+	//
+	// considering cron's minimum granularity is a minute, the seconds range should work fine. defaults to 10s
+	scheduleForwardDur time.Duration
```

Comment:
Is this the lead time for scheduling runs a little before they need to start processing to have ample time to handle cron processing and scheduling the run? If so, should we rename it to lead time to ensure it's the same naming convention we applied previously?